lib/checkbufferoverrun.cpp

```diff
@@ -884,7 +884,13 @@ void CheckBufferOverrun::argumentSizeError(const Token *tok, const std::string &
 // CTU..
 //---------------------------------------------------------------------------
 
-namespace {
+// a Clang-built executable will crash when using the anonymous MyFileInfo later on - so put it in a unique namespace for now
+// see https://trac.cppcheck.net/ticket/12108 for more details
+namespace
+#ifdef __clang__
```

Comment:
It is just a temporary very local workaround. We should not overthink it.

Reply:
sounds good that it's temporary.. however when would it be safe to get rid of this? I assume you used some released clang binary. does it affect many versions?

Reply:
I don't know yet.

Reply:
It affects the clang binaries on ubuntu, so we will probably need to support this for a few years. I think what @danmar is suggesting could be a good idea, perhaps something more like so we dont have to add `using` statements:

```cpp
#ifdef __clang__
#define CLANG_NAMESPACE(NAME) inline namespace NAME ## _internal
#else
#define CLANG_NAMESPACE(NAME) namespace
#endif
```

This should also make it easier to flip on or off for testing purposes.

Reply:
It might also be resolved with some code generation flag which makes the changes unnecessary.

I will do the inline namespaces but I do not want this to spill outside those files. Also that would complicate testing as for easier testing I need to disable it just in a single occurrence and not all of them. So factoring it out doesn't help at all.

The current code have just been merged but now I have to do another rounds of tests to see if that actually works instead of actually working on reporting this upstream...

Reply:
Interesting. It *is* some kind of ODR violation. If I only move one of the object back into the anonymous namespace it works.

Maybe this is actually invalid code/UB which simply isn't detected by anything yet.

Reply:
> Maybe this is actually invalid code/UB which simply isn't detected by anything yet.

interesting! Very good catch if you find it :-)

Reply:
Can we merge this now? This is a release blocker.

Reply:
> If I only move one of the object back into the anonymous namespace it works.

If we are using an anonymous namespace then it shouldn't ODR violation. Probably something is wrong in the linker(or a combination of clang and gnu linker since its only a problem on linux and not mac).

Reply:
> Can we merge this now? This is a release blocker.

Yea I think we should merge this for the release otherwise this cant be built with clang on ubuntu.

Reply:
> If we are using an anonymous namespace then it shouldn't be ODR violation(as they are different symbol names). Probably something is wrong in the linker(or a combination of clang and gnu linker since its only a problem on linux and not mac).

Good point. I want to break down my upstream report in a few days and check the generated symbols. GCC is generating more in a binary than Clang but on that level there is still too much noise in the output.

Feel free to chime in the upstream discussion (if any should occur).